docs/site/Parsing-requests.md

```diff
@@ -260,6 +269,126 @@ A few tips worth mentioning:
   [`api()`](Decorators.md#api-decorator), this requires you to provide a
   completed request body specification.
 
+#### Adding Parsers
+
+Request body parsers can be extended to support various media types. To add a
+new body parser, follow the steps below:
+
+1. Define a class that implements the `BodyParser` interface:
+
+```ts
+/**
+ * Interface to be implemented by body parser extensions
+ */
+export interface BodyParser {
+  /**
+   * Name of the parser
+   */
+  name: string;
+  /**
+   * Indicate if the given media type is supported
+   * @param mediaType Media type
+   */
+  supports(mediaType: string): boolean;
+  /**
+   * Parse the request body
+   * @param request http request
+   */
+  parse(request: Request): Promise<RequestBody>;
+}
+```
+
+See an example at
+https://github.com/strongloop/loopback-next/blob/master/packages/rest/src/body-parsers/body-parser.json.ts.
+
+2. Bind the body parser class to your REST server/application:
+
+For example,
+
+```ts
+server.bodyParser(XmlBodyParser);
+```
+
+The `bodyParser` api binds `XmlBodyParser` to the context with:
+
+- key: `request.bodyParser.XmlBodyParser`
+- tag: `request.bodyParser`
+
+#### Specify Custom Parser by Controller Methods
+
+In some cases, a controller method wants to handle request body parsing by
+itself, such as, to accept `multipart/form-data` for file uploads or stream-line
+a large json document. To bypass body parsing, the `'x-parser'` extension can be
+set to `'stream'` for a media type of the request body content. For example,
+
+```ts
+class FileUploadController {
+  async upload(
+    @requestBody({
+      description: 'multipart/form-data value.',
+      required: true,
+      content: {
+        'multipart/form-data': {
+          // Skip body parsing
+          'x-parser': 'stream',
+          schema: {type: 'object'},
+        },
+      },
+    })
+    request: Request,
```

Comment:
I would keep `request: Request` as is. Most express middleware expects `Request`.

Reply:
It leaves less maneuvering space for us in the future when the controller methods are expecting a full request object instead of a readable stream.

Note that controllers wishing to receive the full request can already do that via `RestBindings.Http.Request`.